euphoria-spark/src/main/java/cz/seznam/euphoria/spark/KeyedWindow.java

```diff
@@ -66,4 +68,17 @@ public String toString() {
             ", key=" + key +
             '}';
   }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public int compareTo(KeyedWindow<W, K> o) {
+    Preconditions.checkArgument(
+        o.key instanceof Comparable,
+        "Key is not comparable, this is probably a bug in euphoria-spark translation.");
```

Comment:
I think this check should be removed in final implementation, maybe so FIXME here?

Reply:
I think it should stay there, as it is illegal state, which means KeyedWindow was used the way it was not intended.

Reply:
I'm generally reserved to testing preconditions in methods of this kind, because they are called **very** often and the check either fails every time or not at all. The performance penalty might be negligible, but it creates sort of "code spam". Misuse of the object should be IMHO covered by tests.